superset-frontend/plugins/plugin-chart-table/src/buildQuery.ts

```diff
@@ -120,7 +120,7 @@ const buildQuery: BuildQuery<TableChartFormData> = (
 
     const moreProps: Partial<QueryObject> = {};
     const ownState = options?.ownState ?? {};
-    if (formDataCopy.server_pagination) {
+    if (formDataCopy.result_type !== 'full' && formDataCopy.server_pagination) {
```

Comment:
I think this should check for the FF `ALLOW_FULL_CSV_EXPORT`

Reply:
Agreed... if it doesn't, and your table has billions of rows, your DB is going to have a bad day.

Reply:
The code above works, but it breaks Explore view (where you edit charts), for some reason Superset tries to load full table to your browser (million rows in my case).
So I tested that this code works well:
`     if (formDataCopy.result_format == 'json' && formDataCopy.server_pagination) {`